### PR TITLE
Show the multiple clauses for the same callback on IEx.Introspection.b/1

### LINKED ISSUE #6546: Multiple @callback with the same name/arity are discarded
### Environment

* Elixir & Erlang versions (elixir --version): 1.6.0-dev, OTP 20
* Operating system: macOS

### Current behavior

Given the following module:

```elixir
defmodule Test do
  @callback test(:foo) :: String.t
  @callback test(:bar) :: [String.t]
end
```

When later getting callback info from iex, we only get the last clause:

```elixir
iex(1)> b Test
@callback test(:bar) :: [String.t()]
```

### Expected behavior

I'm not 100% sure what the expected behaviour is, two possible solutions I see are:

* return all the clauses in `b`.
* warn/raise when defining multiple callbacks like that in the module.

### PR DESCRIPTION
**Pull request by kelvinst**
Fixes #6546

### CODE REVIEW

**Comment by kelvinst** [comment]
Well, I've added this tests and 🎉! We actually compile it correctly, the problem is in `IEx.Introspection.b/1` behaviour that does not show all the clauses of a single callback.

File: lib/iex/lib/iex/introspection.ex [hunk 1]
```diff
@@ -410,15 +410,19 @@ defmodule IEx.Introspection do
...
-    {_, [spec | _]} = List.keyfind(callbacks, key, 0)
+  defp print_callback_docs(name, kind, doc, key, callbacks, printer) do
+    {_, specs} = List.keyfind(callbacks, key, 0)
 
-    definition =
-      Typespec.spec_to_ast(name, spec)
-      |> Macro.prewalk(&drop_macro_env/1)
-      |> Macro.to_string
+    specs
+    |> Enum.reverse()
```
Comment (kelvinst) [inline lib/iex/lib/iex/introspection.ex]: The multiple callback clauses come the reverse of their definition order, so I reverse it to show.

**Comment by fishcakez** [comment]
Could we also get a dialyzer test to verify tooling will treat our callbacks correctly. I think the same callback module as your existing test with 2 implementation modules, one implementation each clause would be good. See https://github.com/elixir-lang/elixir/blob/75150ec7eb7bb373fdf3c188eb01f0c002035cfe/lib/elixir/test/elixir/kernel/dialyzer_test.exs.

**Comment by kelvinst** [comment]
> Could we also get a dialyzer test to verify tooling will treat our callbacks correctly.

I've added a commit, is that test sufficient?

**Comment by fishcakez** [comment]
Yes dialyzer test looks good

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/iex/lib/iex/introspection.ex (hunk 1 above)
Comment (josevalim) [reply lib/iex/lib/iex/introspection.ex]: Oh, good catch. I have pushed a later commit that ensures we write the original order to the beam file.

**Pull request merged**